sympy/core/function.py

```diff
@@ -1230,13 +1230,26 @@ def __new__(cls, expr, *variables, **kwargs):
             if kwargs.get('sort', True):
                 variable_count = cls._sort_variable_count(variable_count)
 
+        # denesting might be possible
+        if isinstance(expr, Derivative):
+            denest = True
+            e = expr.expr
+            old_vc = list(expr.variable_count)
+            for i, (vi, ci) in enumerate(old_vc):
+                for vj, cj in variable_count:
+                    if vi != vj and vi.has(vj):
+                        denest = False
+                        break
+                if not denest:
+                    break
```

Comment:
Maybe this should be
```
if denest:
    variable_count = list(expr.variable_count) + variable_count
    expr = expr.expr
else:
    return expr.func(expr.expr, *variable_count, **kwargs)
```

The first correction of U5 was better than the last one. What should `Derivative(f(g(x),h(x)).diff(g(x),h(x)).diff(x)` give?

Reply:
U5 is back to its compact form